optuna/storages/cached_storage.py

```diff
@@ -0,0 +1,328 @@
+import copy
+import threading
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from optuna import distributions
+from optuna.storages import base
+from optuna.storages.rdb.storage import RDBStorage
+from optuna.study import StudyDirection
+from optuna.study import StudySummary
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+
+
+class _TrialUpdates:
+    def __init__(self) -> None:
+        self.state = None  # type: Optional[TrialState]
+        self.value = None  # type: Optional[float]
+        self.values = dict()  # type: Dict[int, float]
+        self.user_attrs = dict()  # type: Dict[str, Any]
+        self.system_attrs = dict()  # type: Dict[str, Any]
+        self.params = dict()  # type: Dict[str, Any]
+        self.distributions = dict()  # type: Dict[str, distributions.BaseDistribution]
+
+
+class _StudyInfo:
+    def __init__(self) -> None:
+        self.trials = dict()  # type: Dict[int, FrozenTrial]
```

Comment:
This is not a strong opinion but I like `running_trials` as the field name.
(you can ignore this comment)

Reply:
Thank you for the comment. I agree with that, but I'm planning to include finished trials into _StudyInfo in the next PR. So let me keep the name.

Reply:
I see. Thank you for your response.